homeassistant/components/control4/binary_sensor.py

```diff
@@ -0,0 +1,177 @@
+"""Platform for Control4 Alarm Control Panel."""
+from datetime import timedelta
+import json
+import logging
+
+from pyControl4.error_handling import C4Exception
+
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASS_DOOR,
+    DEVICE_CLASS_MOTION,
+    DEVICE_CLASS_OPENING,
+    DEVICE_CLASS_WINDOW,
+    BinarySensorEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_SCAN_INTERVAL
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from . import Control4Entity, get_items_of_category
+from .const import CONF_DIRECTOR, CONTROL4_ENTITY_TYPE, DOMAIN
+from .director_utils import director_update_data
+
+_LOGGER = logging.getLogger(__name__)
+
+CONTROL4_CATEGORY = "sensors"
+CONTROL4_CONTROL_TYPE = "control4_contactsingle"
+CONTROL4_SENSOR_VAR = "ContactState"
+
+CONTROL4_DOOR_PROXY = "contactsingle_doorcontactsensor_c4"
+CONTROL4_WINDOW_PROXY = "contactsingle_windowcontactsensor_c4"
+CONTROL4_MOTION_PROXY = "contactsingle_motionsensor_c4"
+
+CONTROL4_PROXY_MAPPING = {
+    CONTROL4_DOOR_PROXY: DEVICE_CLASS_DOOR,
+    CONTROL4_WINDOW_PROXY: DEVICE_CLASS_WINDOW,
+    CONTROL4_MOTION_PROXY: DEVICE_CLASS_MOTION,
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities
+):
+    """Set up Control4 alarm control panels from a config entry."""
+    entry_data = hass.data[DOMAIN][entry.entry_id]
+    scan_interval = entry_data[CONF_SCAN_INTERVAL]
+    _LOGGER.debug(
+        "Scan interval = %s", scan_interval,
+    )
+
+    async def async_update_data():
+        """Fetch data from Control4 director for alarm control panels."""
+        try:
+            return await director_update_data(hass, entry, CONTROL4_SENSOR_VAR)
```

Comment:
Would it make sense to use just one updated and add `CONTROL4_SENSOR_VAR` to the list of vars?